packages/plugins/response-cache/src/index.ts

```diff
@@ -22,35 +30,105 @@ function sessionFactoryForEnvelop({ request }: YogaInitialContext) {
   return operationIdByRequest.get(request)
 }
 
+const cacheKeyFactoryForEnvelop: BuildResponseCacheKeyFunction =
+  async function cacheKeyFactoryForEnvelop({ sessionId }) {
+    if (sessionId == null) {
+      throw new Error(
+        '[useResponseCache] This plugin is not configured correctly. Make sure you use this plugin with GraphQL Yoga',
+      )
+    }
+    return sessionId
+  }
+
+const getDocumentStringForEnvelop: GetDocumentStringFunction = (
+  executionArgs,
+) => {
+  const context = executionArgs.contextValue as YogaInitialContext
+  if (context.params?.query == null) {
+    throw new Error(
+      '[useResponseCache] This plugin is not configured correctly. Make sure you use this plugin with GraphQL Yoga',
+    )
+  }
+  return context.params.query as string
+}
+
 export function useResponseCache(options: UseResponseCacheParameter): Plugin {
   const buildResponseCacheKey: BuildResponseCacheKeyFunction =
     options?.buildResponseCacheKey || defaultBuildResponseCacheKey
   const cache = options.cache ?? createInMemoryCache()
   const enabled = options.enabled ?? (() => true)
+  const cachedLastModifiedByRequest = new WeakMap<Request, string>()
+  let logger: YogaLogger
   return {
+    onYogaInit({ yoga }) {
+      logger = yoga.logger
+    },
     onPluginInit({ addPlugin }) {
       addPlugin(
         useEnvelopResponseCache({
           ...options,
           cache,
+          getDocumentString: getDocumentStringForEnvelop,
           session: sessionFactoryForEnvelop,
+          buildResponseCacheKey: cacheKeyFactoryForEnvelop,
+          shouldCacheResult({ result }) {
+            const shouldCached = options.shouldCacheResult
+              ? options.shouldCacheResult({ result })
+              : !result.errors?.length
+            if (shouldCached) {
+              result.extensions ||= {}
+              result.extensions.responseCacheLastModified =
+                new Date().toString()
```

Comment:
Any reason why we can not add this in the other extensions response cache key?

Reply:
Envelop plugin doesn't respect existing properties :/
like this;
https://github.com/n1ru4l/envelop/blob/main/packages/plugins/response-cache/src/plugin.ts#L315

Reply:
Should we fix this in envelop, by also spreading the extensions for responseCache?

Reply:
Yes this should be fixed there and I also think we need more work in here to avoid other tricks we do to satisfy Envelop plugin. Maybe this shouldn't extend the Envelop plugin but instead we should make this a standalone plugin by re-useing some logic from there.

Reply:
Okay in private chat we decided that we will drop this from the extensions for the end-user as we only need to use it internally.